pkg/controller/generic/qtransform/qtransform.go

```diff
@@ -237,7 +247,15 @@ func (ctrl *QController[Input, Output]) reconcileRunning(ctx context.Context, lo
 			return nil
 		}
 
-		return err
+		if requeueError == nil || requeueError.Err() == nil {
```

Comment:
this is still incorrect I think. 

```go
if requeueError != nil && requeueError.Err() == err` {
   return requeueError
}

return err
```

Reply:
yes, now I read it and it finally clicked for me :D changed.